src/genn/backends/opencl/presynapticUpdateStrategy.cc

```diff
@@ -125,6 +121,10 @@ void PreSpan::genCode(CodeStream &os, const ModelSpecMerged &modelMerged, const
 
             std::string code = threshOsStream.str();
 
+            if (!wu->getSimSupportCode().empty()) {
```

Comment:
I think this belongs in [this function](https://github.com/genn-team/genn/blob/opencl/src/genn/genn/code_generator/generateSynapseUpdate.cc#L113) like the neuron stuff (you might need to add the namespace as a parameter or something).

Reply:
Do you mean we should just substitute the same way but in `applySynapseSubstitutions()` before the `synapseSubs.apply(code);` call so we don't repeat the code here in `presynapticUpdateStrategy`?

If so, then I didn't do that because the backend variable is not available there to check if the backend does not support namespace. Although we can pass the backend itself or a boolean (`backend.supportsNamepsace()`) to `applySynapseSubstitutions()` from [here](https://github.com/genn-team/genn/blob/opencl/src/genn/genn/code_generator/generateSynapseUpdate.cc#L172).

Is that it?

Reply:
Yeah, exactly. Feel free to change that function however seems best.